packages/synapse-react-client/src/lib/containers/widgets/query-filter/FacetFilterControls.tsx

```diff
@@ -195,77 +262,78 @@ export function FacetFilterControls(props: FacetFilterControlsProps) {
-            )
-          })}
+      {(facets ?? [])
+        .filter(facet => facetFiltersShown.has(facet.columnName))
+        .map(facet => {
+          const columnModel = columnModels!.find(
+            model => model.name === facet.columnName,
+          )
+          return (
+            <div className="FacetFilterControls__facet" key={facet.columnName}>
+              {facet.facetType === 'enumeration' && columnModel && (
+                <EnumFacetFilter
+                  containerAs="Collapsible"
+                  collapsed={false}
+                  facetValues={facet.facetValues}
+                  columnModel={columnModel}
+                  onChange={(facetNamesMap: Record<string, string>) =>
+                    applyMultipleChangesToValuesColumn(
+                      lastRequest,
+                      facet,
+                      applyChanges,
+                      facetNamesMap,
+                    )
+                  }
+                  onClear={() =>
+                    applyChangesToValuesColumn(lastRequest, facet, applyChanges)
+                  }
+                ></EnumFacetFilter>
+              )}
+              {facet.facetType === 'range' && columnModel && (
+                <RangeFacetFilter
+                  facetResult={facet}
+                  columnModel={columnModel}
+                  collapsed={false}
+                  onChange={(values: (string | number | undefined)[]) =>
+                    applyChangesToRangeColumn(
+                      lastRequest,
+                      facet,
+                      applyChanges,
+                      values as string[],
+                    )
+                  }
+                ></RangeFacetFilter>
+              )}
+            </div>
+          )
+        })}
       <div>
         <div className="AvailableFacet">
           <label className="AvailableFacet__label">Available Facets</label>
         </div>
-        {(facets ?? [])
-          .filter(facet => shownChips.includes(facet.columnName))
-          .map(facet => {
-            return (
-              <FacetChip
-                key={facet.columnName}
-                facet={facet}
-                onClick={() => onClick(facet)}
-                isChecked={facetFiltersShown.includes(facet.columnName)}
-              />
-            )
-          })}
+        {facets.map(facet => {
+          return (
+            <FacetChip
+              key={facet.columnName}
+              facet={facet}
+              onClick={() => toggleShowFacetFilter(facet)}
+              isChecked={facetFiltersShown.has(facet.columnName)}
+            />
+          )
+        })}
       </div>
     </div>
   )
 }
+
+export default function FacetFilterControlsOrLoader(
+  props: FacetFilterControlsProps,
+) {
+  const { data, isLoadingNewBundle } = useQueryContext()
+  if (isLoadingNewBundle) {
+    return <FacetFilterControlsSkeleton />
+  } else if (data == null || data.facets == null) {
+    return <></>
+  }
```

Comment:
This matches the existing behavior, but I don't know if it can actually happen. I don't think this component is rendered if there are no facets.